try-slang.js

```diff
@@ -269,6 +388,275 @@ function checkShaderType(userSource)
         return SlangCompiler.PRINT_SHADER;
 }
 
+function parseCommand(command)
+{
+    const match = command.match(/(\w+)\((.*)\)/);
+    if (match)
+    {
+        const funcName = match[1];
+
+        // TODO: This isn't a very robust parser.. any commas in the url will break it.
+        const args = match[2].split(',').map(arg => arg.trim());
+
+        if (funcName === "ZEROS")
+        {
+            return { type: "ZEROS", size: args.map(Number) };
+        }
+        else if (funcName === "URL")
+        {
+            // remove the quotes
+            const validURLMatch = args[0].match(/"(.*)"/)
+            if (!validURLMatch)
+            {
+                throw new Error(`Invalid URL: ${url}`);
+            }
+            
+            return { type: "URL", url: validURLMatch[1] };
+        }
+        else if (funcName === "RAND")
+        {
+            return { type: "RAND", size: args.map(Number) };
+        };
+
+    }
+    else
+    {
+        throw new Error(`Invalid command: ${command}`);
+    }
+}
+
+function parseResourceCommands(userSource)
+{
+    // Now we'll handle some special comments that the user can provide to initialize their resources.
+    //
+    // Here are some patterns we support:
+    // 1. //! @outputBuffer: ZEROS(512, 512)   ==> Initialize "outputBuffer" with zeros of the provided size.
+    // 2. //! @myBuffer: URL("https://example.com/image.png", 512, 512)   ==> Initialize "myBuffer" with image from URL.
+    // 3. //! @noiseBuffer: RAND(1000)   ==> Initialize "myBuffer" with random numbers of the provided size.
```

Comment:
Can you please mentioned about the distribution of the random numbers?

Reply:
Added a comment!